### PR TITLE
New features

### PR DESCRIPTION
**Pull request by jdluzen**
Hey Gonzalo,

here's some updates. Let me know how things look.

Thanks,
Joe

### CODE REVIEW

File: PaymentTest/Program.cs [hunk 1]
```diff
@@ -24,12 +25,18 @@ namespace PaymentTest {
     class Program {
         static void Main (string [] args)
         {
-            StripePayment payment = new StripePayment ("YOURSECRETKEYGOESHERE");
-            TestSimpleCharge (payment);
-            TestCustomer (payment);
-            TestCustomerAndCharge (payment);
-            TestGetCharges (payment);
-            TestGetCustomers (payment);
+            StripePayment payment = new StripePayment ("vtUQeOtUnYr7PGCLQ96Ul4zqpDUO4sOE");
```
Comment (gonzalop) [inline PaymentTest/Program.cs:29]: Is that an actual test key?

File: XamarinStripe/StripeSubscriptionStatus.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+using System;
+
+namespace Xamarin.Payments.Stripe {
+    public enum StripeSubscriptionStatus {
+        Unknown,
+        Trialing,
+        Active,
+        Past_Due,
```
Comment (gonzalop) [inline XamarinStripe/StripeSubscriptionStatus.cs]: s/Past_Due/PastDue/ ?

File: XamarinStripe/StripeDuration.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System;
+
+namespace Xamarin.Payments.Stripe {
+    public enum StripeDuration {
```
Comment (gonzalop) [inline XamarinStripe/StripeDuration.cs]: Perhaps StripeCouponDuration is a better name

File: XamarinStripe/StripeInterval.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System;
+
+namespace Xamarin.Payments.Stripe {
+    public enum StripeInterval {
```
Comment (gonzalop) [inline XamarinStripe/StripeInterval.cs]: How about StripePlanInterval?

**Comment by gonzalop** [comment]
It looks awesome! :-)

I have added a couple of inline comments, but even without those cosmetics changes I will merge these changes if you want.

As a side comment, have you tried partial refunds? They work now and add at least 2 properties to the 'charge' object (refunded and amount_refunded).

File: PaymentTest/Program.cs (hunk 1 above)
Comment (jdluzen) [inline PaymentTest/Program.cs:29]: Yes, a public key from their docs.

File: XamarinStripe/StripeSubscriptionStatus.cs (hunk 2 above)
Comment (jdluzen) [inline XamarinStripe/StripeSubscriptionStatus.cs]: Past_Due in the JSON, but I agree, it should be prettier.

**Comment by jdluzen** [comment]
Thanks Gonzalo!

I'll switch those few things, and check on the partial refunds.

**Pull request merged**